pkg-manager/core/src/install/extendInstallOptions.ts

```diff
@@ -151,6 +152,7 @@ export interface StrictInstallOptions {
   hoistWorkspacePackages?: boolean
   virtualStoreDirMaxLength: number
   peersSuffixMaxLength: number
+  modifyBinPaths?: (manifest: ProjectManifest, extraBinPaths: string[]) => string[] | Promise<string[]>
```

Comment:
I don't think this is a good name for the function as it doesn't just modify the bin paths. It also downloads the node.js version that will be used during execution.

Maybe calling it `prepareExecutionEnv` would be better.

Also, why can the result be sync?

Reply:
also, I don't think the whole manifest should be passed, just the node version

Reply:
> Also, why can the result be sync?

It will call `getNodeBinDir` which is async.

Reply:
https://github.com/pnpm/pnpm/pull/8277/commits/b6cb593e8bfbee5b86bed25d9f0d8e2b11d1ecf5